auditbeat/module/file_integrity/event.go

```diff
@@ -348,14 +362,86 @@ func buildMetricbeatEvent(e *Event, existedBefore bool) mb.Event {
 			errors[idx] = err.Error()
 		}
 		if n == 1 {
-			out.MetricSetFields.Put("error.message", errors[0]) //nolint:errcheck // Will not error.
+			out.MetricSetFields.Put("error.message", errors[0])
 		} else {
-			out.MetricSetFields.Put("error.message", errors) //nolint:errcheck // Will not error.
+			out.MetricSetFields.Put("error.message", errors)
 		}
 	}
 	return out
 }
 
+func aclText(s string) ([]string, error) {
+	b, err := base64.StdEncoding.DecodeString(strings.TrimPrefix(s, "0s"))
```

Comment:
I've kept the `POSIXACLAccess` field as a `string`. It sits right on the boundary of making a decision to change it to a `[]byte`, but I think it's just slightly better as `string`.

Reply:
The only impact of this decision to users will be the debug message that shows the old and new metadata. How does go encode a `[]byte`? IIRC it uses base64 which IMO would be slightly nicer in the logs.

`{"log.level":"debug","@timestamp":"2023-08-16T10:30:29.388-0400","log.logger":"file_integrity","log.origin":{"file.name":"file_integrity/metricset.go","file.line":296},"message":"File changed since it was last seen","service.name":"auditbeat","file_path":"/foo.bar","took":280205,"event":{"action":"attributes_modified","old":{"timestamp":"2023-08-16T14:30:00.686192244Z","path":"/foo.bar","info":{"inode":71102,"uid":0,"gid":0,"sid":"","owner":"","group":"","size":0,"mtime":"2023-08-14T20:49:03.249161413Z","ctime":"2023-08-15T17:49:52.337561405Z","type":"file","mode":444,"setuid":false,"setgid":false,"origin":null,"selinux":"system_u:object_r:quota_db_t:s0","posix_acl_access":"\u0002\u0000\u0000\u0000\u0001\u0000\u0006\u0000\ufffd\ufffd\ufffd\ufffd\u0002\u0000\u0007\u0000o\u0000\u0000\u0000\u0004\u0000\u0004\u0000\ufffd\ufffd\ufffd\ufffd\u0010\u0000\u0007\u0000\ufffd\ufffd\ufffd\ufffd \u0000\u0004\u0000\ufffd\ufffd\ufffd\ufffd"},"source":"scan","action":"none","hash":{"sha1":"da39a3ee5e6b4b0d3255bfef95601890afd80709"}},"new":{"timestamp":"2023-08-16T14:30:29.388513038Z","path":"/foo.bar","info":{"inode":71102,"uid":0,"gid":0,"sid":"","owner":"root","group":"root","size":0,"mtime":"2023-08-14T20:49:03.249161413Z","ctime":"2023-08-16T14:30:29.384200342Z","type":"file","mode":444,"setuid":false,"setgid":false,"origin":null,"selinux":"system_u:object_r:quota_db_t:s0","posix_acl_access":"\u0002\u0000\u0000\u0000\u0001\u0000\u0006\u0000\ufffd\ufffd\ufffd\ufffd\u0002\u0000\u0003\u0000o\u0000\u0000\u0000\u0004\u0000\u0004\u0000\ufffd\ufffd\ufffd\ufffd\u0010\u0000\u0007\u0000\ufffd\ufffd\ufffd\ufffd \u0000\u0004\u0000\ufffd\ufffd\ufffd\ufffd"},"source":"fsnotify","action":"attributes_modified","hash":{"sha1":"da39a3ee5e6b4b0d3255bfef95601890afd80709"}},"ecs.version":"1.6.0"}}`

Reply:
We could add a MarshalText method to the `POSIXACLAccess` field that renders it in the standard format. WDYT?